test/test_associations.rb

```diff
@@ -367,7 +367,21 @@ def test_foreign_key_present_with_null_association_ids
     assert_equal(false, associations.send('foreign_key_present?'))
   end
 
-  def test_ids_equals_for_non_CPK_case
+  def test_assignment_by_ids_as_arrays
+    room = Room.create dorm: dorms(:branner), room_id: 4
+    room.room_assignment_ids = [[3, 1, 2], [4, 1, 2]]
+    room.save!
+    assert_equal room.room_assignments.map(&:student_id), [3, 4]
+  end
+
+  def test_assignment_by_ids_as_strings
+    room = Room.create dorm: dorms(:branner), room_id: 4
+    room.room_assignment_ids = ['3,1,2', '4,1,2']
+    room.save!
+    assert_equal room.room_assignments.map(&:student_id), [3, 4]
+  end
+
+  def test_assignment_by_ids_for_non_CPK_case
```

Comment:
I've slightly changed the name for this test case since I felt the word `equals` might be somewhat confusing with the assertion's 'equal', but I don't have strong opinion on this.